src/renderers/shared/fiber/ReactFiberDevToolsHook.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactFiberDevToolsHook
+ * @flow
+ */
+
+/* globals __REACT_DEVTOOLS_GLOBAL_HOOK__ */
+
+'use strict';
+
+import type { Fiber } from 'ReactFiber';
+import type { FiberRoot } from 'ReactFiberRoot';
+
+var warning = require('warning');
+
+let rendererID = null;
+if (
+  typeof __REACT_DEVTOOLS_GLOBAL_HOOK__ !== 'undefined' &&
+  typeof __REACT_DEVTOOLS_GLOBAL_HOOK__.supportsFiber
+) {
+  exports.injectInternals = function(internals : Object) {
+    warning(rendererID == null, 'Cannot inject into DevTools twice.');
+    rendererID = __REACT_DEVTOOLS_GLOBAL_HOOK__.inject({
+      ...internals,
+      // Increment when internals become incompatible with DevTools.
+      // This way older DevTools can ignore newer Fiber versions.
+      version: 1,
```

Comment:
I'm not sure it makes sense to put the version here. 1) It adds runtime `Object.assign` cost and more code. Probably resulting in an unoptimized object atm. 2) The version indicates what API is exposed, but that API is defined by the DOM renderer here. If you bump this you have to make sure that all consumers and downstream renderers bump theirs too. So isn't the current version more associated with the DOM renderer rather than this shared helper?

I'm also skeptical about version numbers being super useful here anyway. Often we'll want to add features to the devtools that work with minor already released versions without the ability to bump this so we would and those features might not work with the whole range of the version.

So I think we'll probably just end up with feature detection anyway.